### ISSUE TITLE
Live stream crashes after new widevine  update 

### ISSUE THREAD

**Issue by ajay01994** [peak3d/inputstream.adaptive#1519]
Describe the bug

After updating Widevine, the live stream of channels on Kodi starts to stop and crash unexpectedly. This issue began occurring after the update. Before the update, streaming was smooth and without interruptions. The problem seems to be related to the Widevine update as it coincides with the onset of the crashes.

Expected Behavior

The expected behavior is for live streams to play smoothly without interruptions or crashes, as they did before the Widevine update.

Actual Behavior

The actual behavior is that live streams start as expected but then suddenly stop and crash. This disrupts viewing and requires restarting Kodi, only for the issue to occur again after some time.

Possible Fix

A possible fix could involve reverting to a previous version of Widevine or looking for an update or patch that addresses this issue.

To Reproduce

Open Kodi version 20.3 on LibreELEC 11.0.6 running on a Raspberry Pi 4B.
Start any live stream channel.
Observe the stream for a few minutes.
The stream stops and Kodi crashes unexpectedly.
Debuglog
The debuglog can be found here: [Insert debug log link]

MPD/M3U8s/ISM
An example or copy of a manifest (or manifests for HLS - master and variants) can be found here: [Insert manifest link]

Screenshots
Here are some links or screenshots to help explain the problem: [Insert screenshot link]

Additional context or screenshots (if appropriate)

This bug has significantly affected the live streaming experience on Kodi, making it nearly impossible to watch live channels for extended periods without interruptions. The issue started immediately after updating Widevine, suggesting a potential compatibility issue with the new version.

Your Environment
Used Operating system:

 Android

 iOS

 tvOS

 Linux

 OSX

 Windows

 Windows UWP

Operating system version/name: LibreELEC 11.0.6

Kodi version: 20.3

inputstream.adaptive version: [Insert version]

note: Once the issue is made we require you to update it with new information should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.

**Comment by CastagnaIT** [comment]
Is not enough to understand the problem
please include all this info:
1) have you tried to completely reboot the device? since its know problem that after ISA/Widevine update problems may happens when not rebooted
2) most important, you must provide the LibreELEC stacktrace/log having kodi debug enabled, if you dont know how to get it, read LE wiki or ask on their forum, without it we cannot help
3) what widevine version cause the crash
4) know if by reverting to previous widevine version (and specify what version) playback works or not
5) Since we dont develop directly on Kodi OS such as LE/CE, if possible is good to know if the problem is reproducible on other systems such as Windows or linux

**Comment by ajay01994** [comment]
Hello,

I am experiencing crashes on Kodi 20.3 running on a Linux system with ARM64 architecture after updating Widevine CDM. Here are the details for the troubleshooting steps and the information you requested:

Device Reboot:

Yes, I have fully rebooted the device post-update. The issue persists, indicating it's not a simple initialization problem post-update.
LibreELEC Stack Trace/Log:

I enabled Kodi debugging but need guidance on retrieving the LibreELEC stack trace/log. If there are specific instructions beyond the LE wiki, that would be helpful.
Widevine Version:

The issue started after updating to Widevine version 4.10.2662.3, which was extracted from Chrome OS image HANA with a version number 15699.72.0.
The update was applied on 2024-03-25 at 14:24, as per the details in the screenshot.
Previous Widevine Version:

I have reverted to the previous version of Widevine to see if this resolves the issue. However, the problem still persists. The version I reverted to was Widevine 4.10.2252.0, which was working without issues before the update. I would appreciate any advice on further steps or if there's another specific version I should try downgrading to.
![input stream adaptive version](https://github.com/xbmc/inputstream.adaptive/assets/31235529/50d6ec3d-53cc-4210-88e8-352adb916330)
![widevine version](https://github.com/xbmc/inputstream.adaptive/assets/31235529/bde741ad-6c32-44aa-b5bb-b42597982915)

Reproducibility on Other Systems:

I haven’t tested this on other systems yet. However, I will try to replicate the issue on a Windows/Linux system and update this issue with my findings.

Thank you for your assistance.

**Comment by CastagnaIT** [comment]
> I enabled Kodi debugging but need guidance on retrieving the LibreELEC stack trace/log. If there are specific instructions beyond the LE wiki, that would be helpful.

the logs/stacktrace are stored on LE storage
from what i remember you need to connect your LE device with a pc via network samba or nfs in order to copy them to your pc
but maybe over the time their have implemented a more easy way to get these files
as said above the best place is ask on LibreELEC forum, we collab with LibreELEC but its not our competence

**Comment by CastagnaIT** [comment]
maybe better ask to inputStream helper that manage widevine updates,
@horstle are you aware of some crashes on LE ARM64? can you try help/investigate?

**Comment by horstle** [comment]
Did you do a reboot after downgrading widevine? That's very important.

**Comment by ajay01994** [comment]
Yes i did ...

**Comment by ajay01994** [comment]
i have done all permutations and combinations but it randomly stops ,i am looking if i can provide with logs . Should not have updated wide vine

**Comment by horstle** [comment]
I doubt that it's related to Widevine. If it was, reverting Widevine would have helped.

But a log would definitely help determining what is going on.

**Comment by ajay01994** [comment]
Hi , could you suggest which widvine version should be better and is stable ..from my investigation..while i stream the same channel from youtube it doesn't hang..there were no crashes reported while streaming from youtube

**Comment by CastagnaIT** [comment]
youtube dont use widevine
if no debug stacktrace will be provided in a week i will close the issue
since for us there is no way to understand something
conjecture are useless with crashes

**Comment by ajay01994** [comment]
Hello everyone,

I would like to provide an update on the issue discussed here. After a more detailed review of the system logs, it turns out that the problem was not related to the Widevine update as I initially thought. My apologies for the confusion.

It appears the root cause was a power supply issue. Fluctuations in power voltage coincidentally impacted the Wi-Fi signal, which in turn affected the live stream. I have now ensured that the Raspberry Pi 4B is equipped with the appropriate power supply, and I am happy to report that the issue has been resolved.

I appreciate your patience and assistance as we worked through this. Since the issue is unrelated to Widevine or the project's codebase, and is now resolved with the correct power supply setup, I am going to close this issue now. Thank you all for your help!

Best regards,
